mirage/yaml/yaml_generator.py

```diff
@@ -388,11 +387,12 @@ def create_inputs(self):
         grism_source_image = ['False'] * len(self.info['Mode'])
         grism_input_only = ['False'] * len(self.info['Mode'])
         for i in range(len(self.info['Mode'])):
-            if self.info['Mode'][i] == 'WFSS':
-                grism_source_image[i] = 'True'
-                grism_input_only[i] = 'True'
+            if self.info['Mode'][i].lower() == 'wfss':
+                if self.info['detector'] == 'NIS' or self.info['detector'][i][-1] == '5':
+                    grism_source_image[i] = 'True'
+                    grism_input_only[i] = 'True'
                 # SW detectors shouldn't be wfss
-                if self.info['detector'][i][-1] != '5':
+                if self.info['instrument'] == 'NIRCAM' and self.info['detector'][i][-1] != '5':
```

Comment:
It looks like this line is causing Travis to fail. I think the instrument key is capitalized (`'Instrument'`).

Reply:
Good catch! I've capitalized it in my latest fixes.